typescript/connections.md

```diff
@@ -0,0 +1,151 @@
+# SDK Connection Configuration
+
+## Problem statement
+
+The current API for connecting a Worker to the server is via a `Core` singleton.
+This singleton can only connect to one namespace per process and, by looking at our basic samples,
+it not immediately obvious how to configure the server address for a given Worker.
+
+```ts
+// worker.ts
+await Core.install({ serverOptions: { address, namespace } });
+// Worker uses a connection created by the Core singleton
+const worker = await Worker.create({ taskQueue });
+```
+
+## Proposals
+
+### Explicit `Connection` objects
+
+- `Connection` in `@temporalio/client` for client only
+- `NativeConnection` in `@temporalio/worker` for sharing connections between different workers
+
+```ts
+// worker.ts
+import { NativeConnection, Worker } from '@temporalio/worker';
+
+const connection = await NativeConnection.create({ address });
+const worker = await Worker.create({ connection, namespace, taskQueue });
+
+// starter.ts
+
+// same as today but Connection becomes async to make the `getSystemInfo` call
+import { Connection, WorkflowClient } from '@temporalio/client';
+
+const connection = await Connection.create({ address });
+const client = new WorkflowClient(connection.service, { namespace });
+```
+
+#### Pros
+
+- Explicit connection reuse
+
+#### Cons
+
+- Verbose
+- Confusing when to use each connection type
+
+### Implicit connection reuse with similar options
+
+```ts
+// worker.ts
+import { Worker } from '@temporalio/worker';
+
+// worker1 and worker2 share the same underlying connection based on given options
+const worker1 = await Worker.create({ connectionOptions: { address }, namespace, taskQueue: 'tq1' });
+const worker2 = await Worker.create({ connectionOptions: { address }, namespace, taskQueue: 'tq2' });
+
+// starter.ts
+import { WorkflowClient } from '@temporalio/client';
+
+const client = await WorkflowClient.create({ connectionOptions: { address }, namespace });
+```
+
+#### Pros
+
+- Short
+- Easy path from sample with local server to real world use case
+
+#### Cons
+
+- Implicit connection sharing is non-obvious
+- No way to opt out of connection sharing (can be addressed if needed)
+
+### Mix of options and explicit clients (recommended)
```

Comment:
Why recommended/ And why not list the cons of having multiple ways to create a connection and having the user have to care about connection reuse to get its benefits?

Reply:
I regret recommending it. I want to think about this some more